appmgr/src/net/mdns.rs

```diff
@@ -210,18 +217,30 @@ impl MdnsControllerInner {
         pkg_id: &PackageId,
         interfaces: I,
     ) {
-        self.services.extend(
-            interfaces
-                .into_iter()
-                .map(|(interface_id, key)| ((pkg_id.clone(), interface_id), key)),
-        );
-        self.sync();
+        let mut diff = false;
```

Comment:
Would this be better as an iter().any() combinator?

Reply:
I'd love to tbh. I haven't figured out the way to make that clear. I'll take a stab at it again.